static/src/javascripts/projects/mma-tabs/digitalpack-tab.js

```diff
@@ -0,0 +1,107 @@
+define([
+    'bean',
+    'common/utils/$',
+    'common/utils/ajax',
+    'common/utils/config',
+    'mmaTabs/formatters',
+    'mmaTabs/payment-form'
+], function (bean, $, ajax, config, formatters, PaymentForm) {
+
+    var PACKAGE_COST = '.js-dig-package-cost',
+        PAYMENT_FORM = '.js-dig-card-details',
+        SUBSCRIBER_ID = '.js-dig-subscriber-id',
+        REMAINING_TRIAL_LENGTH = '.js-dig-remaining-trial-length',
+        REMAINING_TRIAL_LENGTH_CONTAINER = '.js-dig-remaining-trial-length-container',
+        PACKAGE_CURRENT_RENEWAL_DATE = '.js-dig-current-renewal-date',
+        PACKAGE_CURRENT_PERIOD_END = '.js-dig-current-period-end',
+        PACKAGE_CURRENT_PERIOD_START = '.js-dig-current-period-start',
+        PACKAGE_NEXT_PAYMENT_DATE = '.js-dig-next-payment-date',
+        PACKAGE_NEXT_PAYMENT_PRICE = '.js-dig-next-payment-price',
+        PACKAGE_NEXT_PAYMENT_CONTAINER = '.js-dig-next-payment-container',
+        PACKAGE_INTERVAL = '.js-dig-plan-interval',
+        DETAILS_JOIN_DATE = '.js-dig-join-date',
+        NOTIFICATION_CANCEL = '.js-dig-cancel-tier',
+        DIGITALPACK_DETAILS = '.js-dig-details',
+        DIGITALPACK_PRODUCT = '.js-dig-product',
+        UP_SELL = '.js-dig-up-sell',
+        DIG_INFO = '.js-dig-info',
+        LOADER = '.js-dig-loader',
+        IS_HIDDEN_CLASSNAME = 'is-hidden',
+        stripeForm = new PaymentForm();
+
+    function fetchUserDetails() {
+        ajax({
+
+            url: config.page.userAttributesApiUrl + '/me/mma-digitalpack',
+            crossOrigin: true,
+            withCredentials: true,
+            method: 'get'
+        }).then(function (resp) {
+            if (resp && resp.subscription) {
+                hideLoader();
+                stripeForm.init($(PAYMENT_FORM)[0], $());
+                populateUserDetails(resp);
+            }
+        }, function (error) {
+            if (error.status == 404) {
+                hideLoader();
+                displayDigitalPackUpSell();
+            }
+        });
+    }
+
+    function hideLoader() {
+        $(LOADER).addClass(IS_HIDDEN_CLASSNAME);
+    }
+
+    /**
+     * @param {{
+     *  optIn,
+     *  joinDate,subscriberId,
+     *  subscription: {trialLength, nextPaymentDate,nextPaymentPrice,renewalDate,
+     *  plan: {name,amount,interval}}
+     * }} userDetails
+     */
+    function populateUserDetails(userDetails) {
+        $(SUBSCRIBER_ID).text(userDetails.subscription.subscriberId);
```

Comment:
As discussed, it would be much nicer to use a templating engine or some library with DOM data-binding or similar to handle all of this. I noticed lodash is already included in the project - this has a [basic templating engine](https://lodash.com/docs#template). Maybe it would be worth giving it a go?